packages/net-stubbing/test/unit/intercepted-request-spec.ts

```diff
@@ -60,5 +62,57 @@ describe('InterceptedRequest', () => {
         mergeChanges: _.merge,
       })
     })
+
+    it('ignores disabled subscriptions', async () => {
+      const socket = {
+        toDriver: sinon.stub(),
+      }
+      const state = NetStubbingState()
+      const interceptedRequest = new InterceptedRequest({
+        req: {
+          matchingRoutes: [
+            // @ts-ignore
+            {
+              id: '1',
+              hasInterceptor: true,
+              routeMatcher: {},
+              disabled: true,
+            },
+            // @ts-ignore
+            {
+              id: '2',
+              hasInterceptor: true,
+              routeMatcher: {},
+            },
+          ],
+        },
+        state,
+        socket,
+      })
+
+      interceptedRequest.addDefaultSubscriptions()
+
+      const data = { foo: 'bar' }
+
+      socket.toDriver.callsFake((eventName, subEventName, frame) => {
+        expect(eventName).to.eq('net:stubbing:event')
+        expect(subEventName).to.eq('before:request')
+        expect(frame).to.deep.include({
+          subscription: {
+            eventName: 'before:request',
+            await: true,
+            routeId: '2',
+          },
+        })
+
+        state.pendingEventHandlers[frame.eventId](frame.data)
```

Comment:
do we need a `done` handler here or something similar in the case the fake for some reason is not called and the test passes when it shouldn't?

Reply:
You can't mix and match promises and done, but I figured a better way to ensure this is getting called: [`339a450` (#30435)](https://github.com/cypress-io/cypress/pull/30435/commits/339a450fe3ade676b440489e52971e0dd0706dd0)